controllers/reconcilers/configuration/prometheus_rules.go

```diff
@@ -111,6 +113,42 @@ func (r *Reconciler) createRequestedRules(cr *v1.Observability, ctx context.Cont
 	return nil
 }
 
+func (r *Reconciler) createDMSAlert(cr *v1.Observability, ctx context.Context) error {
+	// Only create the DMS alert if a custom rule selector is specified
+	if cr.Spec.SelfContained == nil || cr.Spec.SelfContained.RuleLabelSelector == nil {
```

Comment:
Can you add another conditional or here. we are setting `RuleLabelSelector: nil,` We are setting RuleNamespaceSelector instead. So in our case the creation would be skipped.